<a href="https://colab.research.google.com/github/111Vidya/DS-Assignments/blob/main/Hypothesis_Testing_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Q1.A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.
minitabfile : Cutlets.csv

Null Hypothesis Ho - Diameter of the cutlets from both units are SAME.
Alternate Hypothesis H1 - Diameter of the cutlets from both units are NOT SAME

In [ ]:
import pandas as pd
ct=pd.read_csv("Cutlets.csv")
ct.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [ ]:
ct.shape

(35, 2)

In [ ]:
ct.isnull()

Unit A  Unit B
0    False   False
1    False   False
2    False   False
3    False   False
4    False   False
5    False   False
6    False   False
7    False   False
8    False   False
9    False   False
10   False   False
11   False   False
12   False   False
13   False   False
14   False   False
15   False   False
16   False   False
17   False   False
18   False   False
19   False   False
20   False   False
21   False   False
22   False   False
23   False   False
24   False   False
25   False   False
26   False   False
27   False   False
28   False   False
29   False   False
30   False   False
31   False   False
32   False   False
33   False   False
34   False   False

In [ ]:
#Two Sample T Test
import scipy
from scipy import stats

In [ ]:
stats.ttest_ind(ct['Unit A'],ct['Unit B'])

TtestResult(statistic=0.7228688704678063, pvalue=0.4722394724599501, df=68.0)

**ANS : PValue = 0.47 which is greater than 0.05.
PValue > 0.05 accept Null Hypothesis Ho.
Diameter of the cutlets from both units are SAME.**

Q2. A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.       
Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

In [ ]:
lb=pd.read_csv("LabTAT.csv")
lb

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [ ]:
lb.isnull()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0           False         False         False         False
1           False         False         False         False
2           False         False         False         False
3           False         False         False         False
4           False         False         False         False
..            ...           ...           ...           ...
115         False         False         False         False
116         False         False         False         False
117         False         False         False         False
118         False         False         False         False
119         False         False         False         False

[120 rows x 4 columns]

In [ ]:
lb.isnull().sum()

Laboratory 1    0
Laboratory 2    0
Laboratory 3    0
Laboratory 4    0
dtype: int64

In [ ]:
lb.mean()

Laboratory 1    178.361583
Laboratory 2    178.902917
Laboratory 3    199.913250
Laboratory 4    163.682750
dtype: float64

Null Hypothesis Ho - TAT of all Labs are same TATL1=TATL2=TATL3=TATL4.
Alternate Hypothesis H1 - TAT of atleast one of the lab NOT SAME

In [ ]:
stats.f_oneway(lb.iloc[:,0],lb.iloc[:,1],lb.iloc[:,2],lb.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

**ANS: PValue < 0.05 accept Alternate Hypothesis H1.
TAT of atleast one Laboratory is different**

Q3. Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

Null Hypothesis Ho -All Proportions are equal.
Alternate Hypothesis H1 -Not all proportions are equal


In [ ]:
br=pd.read_csv("BuyerRatio.csv")
br

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
br1=br.drop(columns="Observed Values",axis=1)
br1

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [ ]:
Chisquares_results=scipy.stats.chi2_contingency(br1)
Chisquares_results

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

**ANS: PValue > 0.05 accept Null Hypothesis Ho
All Proportions are equal**

Q4. TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

Null Hypothesis Ho-% of defects accross 4 centers is same.
Alternate Hypothesis H1-% of defects accros 4 centers are not same

In [ ]:
orderform=pd.read_csv("Costomer+OrderForm.csv")
orderform

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [ ]:
orderform['India'].value_counts(),orderform['Indonesia'].value_counts(),orderform['Malta'].value_counts(),
orderform['Phillippines'].value_counts()

(Error Free    280
 Defective      20
 Name: India, dtype: int64,
 Error Free    267
 Defective      33
 Name: Indonesia, dtype: int64,
 Error Free    269
 Defective      31
 Name: Malta, dtype: int64,
 Error Free    271
 Defective      29
 Name: Phillippines, dtype: int64)

In [ ]:
orderform1=([[280,267,269,271],[20,33,31,29]])

In [ ]:
Chisquares_results1=scipy.stats.chi2_contingency(orderform1)
Chisquares_results1

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

**ANS: PValue =0.277>0.05 accept Null Hypothesis Ho.
% Defects accross 4 centers are same**